src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -114,34 +115,37 @@ public void RelayWorker()
             this.asyncLoop = this.asyncLoopFactory.Run($"{this.name}.RelayWorker", async token =>
             {
                 this.logger.LogTrace("()");
-                List<uint256> blocks = this.blockHashesToAnnounce.Keys.ToList();
 
-                if (!blocks.Any())
+                if (!this.blockHashesToAnnounce.Any())
                 {
                     this.logger.LogTrace("(-)[NO_BLOCKS]");
                     return;
                 }
 
                 var broadcastItems = new List<uint256>();
-                foreach (uint256 blockHash in blocks)
+
+                while (this.blockHashesToAnnounce.TryPeek(out uint256 blockHash))
                 {
                     // The first block that is not in disk will abort the loop.
                     if (!await this.blockRepository.ExistAsync(blockHash).ConfigureAwait(false))
                     {
-                        // NOTE: there is a very minimal possibility a reorg would happen
-                        // and post reorg blocks will now be in the 'blockHashesToAnnounce',
-                        // current logic will discard those blocks, I suspect this will be unlikely
-                        // to happen. In case it does a block will just not be relays.
+                        // There is a small possibility that a reorg happened and 'blockHashesToAnnounce'
```

Comment:
avoid saying small/minimal possibility/chance
we are in blockchain code, reorgs/races are normal

Reply:
Yeah thats true, I think my initial intention was there is a small possibility a reorg happened in between the block arrived and pending relay.

Lets fix this comment to be
>In cases when the node had a reorg the 'blockHashesToAnnounce' will contain blocks that are not anymore on the main chain.
etc..